packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -863,3 +1141,503 @@ class _NestedOuterBallisticScrollActivity extends BallisticScrollActivity {
+      paintExtent: childLayoutGeometry.paintExtent,
+      paintOrigin: childLayoutGeometry.paintOrigin,
+      layoutExtent: childLayoutGeometry.paintExtent - childLayoutGeometry.maxScrollObstructionExtent,
+      maxPaintExtent: childLayoutGeometry.maxPaintExtent,
+      maxScrollObstructionExtent: childLayoutGeometry.maxScrollObstructionExtent,
+      hitTestExtent: childLayoutGeometry.hitTestExtent,
+      visible: childLayoutGeometry.visible,
+      hasVisualOverflow: childLayoutGeometry.hasVisualOverflow,
+      scrollOffsetCorrection: childLayoutGeometry.scrollOffsetCorrection,
+    );
+    handle._setExtents(childLayoutGeometry.maxScrollObstructionExtent, childLayoutGeometry.maxScrollObstructionExtent);
+  }
+
+  @override
+  void applyPaintTransform(RenderObject child, Matrix4 transform) {
+    // child is always at our origin
+  }
+
+  @override
+  bool hitTestChildren(HitTestResult result, { @required double mainAxisPosition, @required double crossAxisPosition }) {
+    if (child != null)
+      return child.hitTest(result, mainAxisPosition: mainAxisPosition, crossAxisPosition: crossAxisPosition);
+    return false;
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (child != null)
+      context.paintChild(child, offset);
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder description) {
+    super.debugFillProperties(description);
+    description.add(new DiagnosticsProperty<SliverOverlapAbsorberHandle>('handle', handle));
+  }
+}
+
+/// A sliver that has a sliver geometry based on the values stored in a
+/// [SliverOverlapAbsorberHandle].
+///
+/// The [SliverOverlapAbsorber] must be an earlier descendant of a common
+/// ancestor [Viewport], so that it will always be laid out before the
+/// [SliverOverlapInjector] during a particular frame.
+///
+/// See also:
+///
+///  * [NestedScrollView], which uses a [SliverOverlapAbsorber] to align its
+///    children, and which shows sample usage for this class.
+class SliverOverlapInjector extends SingleChildRenderObjectWidget {
+  /// Creates a sliver that is as tall as the value of the given [handle]'s
+  /// layout extent.
+  ///
+  /// The [handle] must not be null.
+  const SliverOverlapInjector({
+    Key key,
+    @required this.handle,
+    Widget child,
+  }) : assert(handle != null),
+       super(key: key, child: child);
+
+  /// The handle to the [SliverOverlapAbsorber] that is feeding this injector.
+  ///
+  /// This should be a handle owned by a [SliverOverlapAbsorber] and a
+  /// [NestedScrollViewViewport].
+  final SliverOverlapAbsorberHandle handle;
+
+  @override
+  RenderSliverOverlapInjector createRenderObject(BuildContext context) {
+    return new RenderSliverOverlapInjector(
+      handle: handle,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, RenderSliverOverlapInjector renderObject) {
+    renderObject
+      ..handle = handle;
```

Comment:
Cascade operator looks kind of weird here if you're only setting one thing.

Reply:
as above